### PR TITLE
build: update to TypeScript 5.5 stable

### PR DESCRIPTION
**Pull request by crisbeto**
Updates the repo to the stable release of TypeScript 5.5.

### CODE REVIEW

**Review by josephperrott — APPROVED** [review]
LGTM

**Review by thePunderWoman — APPROVED** [review]
reviewed-for: public-api

**Review by AndrewKushnir — APPROVED** [review]

**Comment by crisbeto** [comment]
I'm holding this PR back for a bit because we're still talking with Alex if it's the right way to resolve one of the issues that popped up.

**Review by pkozlowski-opensource — APPROVED** [review]
LGTM

Reviewed-for: public-api

File: packages/compiler-cli/src/ngtsc/file_system/src/compiler_host.ts [hunk 1]
```diff
@@ -22,16 +22,7 @@ export class NgtscCompilerHost implements ts.CompilerHost {
   getSourceFile(fileName: string, languageVersion: ts.ScriptTarget): ts.SourceFile | undefined {
     const text = this.readFile(fileName);
     return text !== undefined
-      ? ts.createSourceFile(
```
Comment (crisbeto) [inline packages/compiler-cli/src/ngtsc/file_system/src/compiler_host.ts:25]: The context here is that I had to make this change during the update to the TS 5.5 beta, but it's no longer necessary for the RC/final.

**Review by alxhub — APPROVED** [review]

**Comment by dylhunn** [comment]
This PR was merged into the repository by commit ccc8c80cd034c8f1742ddd43d6ec825f3b9f65bf.

The changes were merged into the following branches: main

**Pull request closed without merging**